weave

```diff
@@ -1041,32 +1037,16 @@ wait_for_status() {
     done
 }
 
-# Parse the output of `ip addr show`, outputing weave addresses of kind $1
-addrs_from_ip_addr_show() {
-    while read first second rest ; do
-        if [ "$first" != "${first%:}" ] ; then
-            iname="${second%:}"
-        elif [ "$first" = "$1" -a \( "$iname" = "weave" -o "$iname" != "${iname#ethwe}" \) ] ; then
-            echo $second
-        fi
-    done
-}
-
 # Call $1 for all containers, passing container ID, all MACs and all IPs
 with_container_addresses() {
     COMMAND=$1
     shift 1
     for CONTAINER_ID in "$@" ; do
-        if [ "$CONTAINER_ID" = "weave:expose" ] ; then
-            ADDRS_CMD="ip addr show dev $BRIDGE"
-        else
-            ADDRS_CMD="with_container_netns $CONTAINER_ID container_ip_addr_show"
-        fi
-        if CONTAINER_ADDRS=$($ADDRS_CMD 2>/dev/null) ; then
-            CONTAINER_MAC=$(echo "$CONTAINER_ADDRS" | addrs_from_ip_addr_show 'link/ether')
-            CONTAINER_IPS=$(echo "$CONTAINER_ADDRS" | addrs_from_ip_addr_show 'inet')
-            if [ -n "$CONTAINER_MAC$CONTAINER_IPS" ] ; then
-                $COMMAND "$CONTAINER_ID" "$CONTAINER_MAC" "$CONTAINER_IPS"
+        if CONTAINER_ADDRS=$(util_op container-addrs $CONTAINER_ID) ; then
```

Comment:
That's not quite what I had in mind. I was thinking of invoking `container-addrs $@` and then looping over the output. That's probably more trouble than it is worth though.